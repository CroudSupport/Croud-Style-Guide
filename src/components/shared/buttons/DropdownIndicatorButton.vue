
<script>
    import CroudDropdownIcon from '../icons/ArrowIcon'
    /**
    * Atomic button that is used to show or hide content
    *
    * *This is a functional component which has no state so does not need to define any props*
    * @see https://vuejs.org/v2/guide/render-function.html#Functional-Components
    *
    * @example ./croud-dropdown-indicator-button.md
    */
    export default {
        functional: true,

        name: 'croud-dropdown-indicator-button',

        render(h, context) {
            return h(CroudDropdownIcon, {
                props: {
                    direction: context.props.collapsed ? 'right' : 'down',
                },
                on: context.listeners,
            })
        },
    }
</script>

<style scoped>

</style>

