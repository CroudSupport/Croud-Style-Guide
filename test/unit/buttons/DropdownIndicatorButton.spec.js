
import Vue from 'vue'
import DropdownIndicatorButton from '../../../src/components/shared/buttons/DropdownIndicatorButton'

const props = {}
const Constructor = Vue.extend({
    render(h) {
        return h(DropdownIndicatorButton, {
            props,
        })
    },
})

const vm = new Constructor().$mount()

describe('DropdownIndicatorButton', () => {
    it('should match the snapshot', () => {
        expect(vm.$el).toMatchSnapshot()
    })

    it('should change direction when collapsed', () => {
        props.collapsed = true
        vm.$forceUpdate()
        vm.$nextTick(() => {
            expect(vm.$el).toMatchSnapshot()
        })
    })
})
