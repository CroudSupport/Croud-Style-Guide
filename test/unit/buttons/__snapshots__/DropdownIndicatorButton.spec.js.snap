// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownIndicatorButton should change direction when collapsed 1`] = `
<i
  class="chevron icon right"
/>
`;

exports[`DropdownIndicatorButton should match the snapshot 1`] = `
<i
  class="chevron icon down"
/>
`;
